### PR TITLE
feat(medusa-react): Add Collection batch (remove, add) endpoints

### LINKED ISSUE #1089: Add `POST/DELETE /admin/collections/:id/products/batch` endpoints to @medusajs/medusa-js and medusa-react
The new batch endpoints introduced by https://github.com/medusajs/medusa/pull/1032 are missing in the client and medusa-react.

### PR DESCRIPTION
**Pull request by richardwardza**
Depends on #1958 
Closes #1089 

Add POST/DELETE /admin/collections/:id/products/batch endpoints to @medusajs/medusa-react

(Not to be merged before #1958 )

I'm not sure how you would like to handle forks and stacked PR's..

### CODE REVIEW

File: packages/medusa-js/src/resources/admin/collections.ts [hunk 1]
```diff
@@ -88,6 +90,36 @@ class AdminCollectionsResource extends BaseResource {
 
     return this.client.request("GET", path, undefined, {}, customHeaders)
   }
+
+  /**
+   * @description Updates products associated with a Product Collection
```
Comment (richardwardza) [inline packages/medusa-js/src/resources/admin/collections.ts:95]: The `description` and `param` description were copied from the API.

File: packages/medusa-react/mocks/handlers/admin.ts [hunk 2]
```diff
@@ -108,6 +108,29 @@ export const adminHandlers = [
...
+        }
+      })
+    )
+  }),
+
+  rest.delete("/admin/collections/:id/products/batch", (req, res, ctx) => {
+    return res(
+      ctx.status(200),
+      ctx.json({
+        id: fixtures.get("product_collection").id,
```
Comment (adrien2p) [inline packages/medusa-react/mocks/handlers/admin.ts:127]: todo: The id can be assigned with `req.params.id`

File: packages/medusa-react/test/hooks/admin/collections/mutations.test.ts [hunk 3]
```diff
@@ -80,3 +82,57 @@ describe("useAdminDeleteCollection hook", () => {
...
+    expect(result.current.data?.response.status).toEqual(200)
+    expect(result.current.data).toEqual(
+      expect.objectContaining({
+        id: fixtures.get("product_collection").id,
+        object: "product-collection",
+        removed_products: remove.product_ids
+      })
+    )
+  })
+})
```
Comment (adrien2p) [inline packages/medusa-react/test/hooks/admin/collections/mutations.test.ts:138]: todo: Missing eol

**Review by adrien2p — APPROVED** [review]
LGTM 🎉 small comments

File: packages/medusa-react/src/hooks/admin/collections/mutations.ts [hunk 4]
```diff
@@ -62,3 +64,60 @@ export const useAdminDeleteCollection = (
...
+  )
+}
+
+/**
+ * Hook returns function for removal of multiple products from a collection.
+ *
+ * @param id - id of the collection from which products will be removed
+ * @param options
+ */
+export const useAdminRemoveProductssFromCollection = (
```
Comment (olivermrbl) [inline packages/medusa-react/src/hooks/admin/collections/mutations.ts:103]: todo: Typo in name
```suggestion
export const useAdminRemoveProductsFromCollection = (
```

File: packages/medusa-react/src/hooks/admin/collections/mutations.ts [hunk 5]
```diff
@@ -62,3 +64,60 @@ export const useAdminDeleteCollection = (
     )
   )
 }
+
+
+/**
+ * Hook returns functions for addition of multiple products to a collection.
```
Comment (olivermrbl) [inline packages/medusa-react/src/hooks/admin/collections/mutations.ts:70]: todo: Minor fix to grammar
```suggestion
 * Hook returns function for adding multiple products to a collection.
```

File: packages/medusa-react/test/hooks/admin/collections/mutations.test.ts [hunk 6]
```diff
@@ -80,3 +82,57 @@ describe("useAdminDeleteCollection hook", () => {
...
+    expect(result.current.data?.collection).toEqual(
+      expect.objectContaining({
+        ...fixtures.get("product_collection"),
+        products: [fixtures.get("product")],
+      })
+    )
+  })
+})
+
+describe("useAdminRemoveProductssFromCollection hook", () => {
```
Comment (olivermrbl) [inline packages/medusa-react/test/hooks/admin/collections/mutations.test.ts:112]: todo: Typo
```suggestion
describe("useAdminRemoveProductsFromCollection hook", () => {
```

File: packages/medusa-react/test/hooks/admin/collections/mutations.test.ts [hunk 7]
```diff
@@ -80,3 +82,57 @@ describe("useAdminDeleteCollection hook", () => {
...
+  test("remove products from a collection", async () => {
+    const remove = {
+      product_ids: [fixtures.get("product").id],
+    }
+
+    const { result, waitFor } = renderHook(
+      () => useAdminRemoveProductssFromCollection(fixtures.get("product_collection").id),
```
Comment (olivermrbl) [inline packages/medusa-react/test/hooks/admin/collections/mutations.test.ts:119]:
```suggestion
      () => useAdminRemoveProductsFromCollection(fixtures.get("product_collection").id),
```

**Review by olivermrbl — CHANGES_REQUESTED** [review]

**Comment by richardwardza** [comment]
Spelling and grammar updated, Thanks Oliver!

**Comment by richardwardza** [comment]
I forgot the test - will update shortly.

**Review by olivermrbl — APPROVED** [review]
LGTM! Thanks again @richardwardza 💪

**Pull request merged**